### PR TITLE
Feature/Add mailbox to supervisor context.

### PR DESCRIPTION
**Pull request by MCPx**
## Description

Allows onCrash handlers to access the rest of the mailbox through the SupervisorContext.

## Related Issue

#80

## Motivation and Context

Currently the parameters passed to onCrash don't provide access to the internal queue of messages to handle the messages waiting in the queue before shutting down the actor.

## How Has This Been Tested?

Test added to the onCrash tests to check that other messages in the queue are accessible when a message ahead of them in the queue crashes.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: test/actor.js [hunk 1]
```diff
@@ -538,6 +538,31 @@ describe('Actor', function () {
...
+        ctx.mailbox.length.should.equal(2);
+        ctx.mailbox[0].message.should.equal('msg1');
+        ctx.mailbox[1].message.should.equal('msg2');
+        return ctx.escalate;
+      };
+      const parent = createSupervisor(system, 'test1');
+      const child = spawn(parent, (state = 0, msg, ctx) => {
+        // Wait for messages to queue up, then fail on msg0
+        if (state + 1 === 1) {
+          delay(100);
```
Comment (ncthbrt) [inline test/actor.js:552]: This function needs to be async with delay being awaited for this to function as expected consistently.

File: lib/actor.js [hunk 2]
```diff
@@ -195,7 +195,8 @@ class Actor {
       name: this.name,
       children: new Map(this.childReferences),
       sender,
-      log: this.log
+      log: this.log,
+      mailbox: this.mailbox.toArray()
```
Comment (ncthbrt) [inline lib/actor.js:199]: Should this not be in the supervision context rather?

File: lib/index.d.ts [hunk 3]
```diff
@@ -66,7 +66,8 @@ declare module 'nact' {
     self: Ref<Msg>,
     name: ActorName,
     children: Map<ActorName, Ref<unknown>>,
-    log: Logger
+    log: Logger,
+    mailbox: Msg[]
```
Comment (ncthbrt) [inline lib/index.d.ts:70]: What should happen if the crash is escalated? Are the messages in the child dropped?

File: test/actor.js (hunk 1 above)
Comment (MCPx) [reply test/actor.js]: Fixed.

File: lib/index.d.ts (hunk 3 above)
Comment (MCPx) [reply lib/index.d.ts]: Moved it out to an intermediate type between SupervisionContext and ActorContext to be shared with afterStop, as discussed.

File: lib/actor.js (hunk 2 above)
Comment (MCPx) [reply lib/actor.js]: Moved it out to an intermediate type between SupervisionContext and ActorContext to be shared with afterStop, as discussed.

File: lib/index.d.ts (hunk 3 above)
Comment (MCPx) [reply lib/index.d.ts]: Added context to afterStop so that it can be handled there across children, as discussed.

**Comment by ncthbrt** [comment]
Not sure why the travis build is no longer reporting, but looks like it succeeded. I'm going to merge this.

**Pull request merged**